README.md

```diff
@@ -1,95 +1,209 @@
+#
+# What's Deep Learning?assistant
+#
+# Yer lookin' fer a treasure trove o' knowledge on Deep Learnin', eh?
+# Alright then, listen close and I'll tell ye about it.
+#
+#Deep Learnin' be a type o' machine learnin' that uses neural networks
+```
 
 ## Local Inference
 
-Would you like to run inference of the Llama 3.1 models locally? So do we! The memory requirements depend on the model size and the precision of the weights. Here's a table showing the approximate memory needed for different configurations:
-
-<table>
-  <tr>
-   <td><strong>Model Size</strong>
-   </td>
-   <td><strong>FP16</strong>
-   </td>
-   <td><strong>FP8</strong>
-   </td>
-   <td><strong>INT4 (AWQ/GPTQ/bnb)</strong>
-   </td>
-  </tr>
-  <tr>
-   <td>8B
-   </td>
-   <td>16 GB
-   </td>
-   <td>8 GB
-   </td>
-   <td>4 GB
-   </td>
-  </tr>
-  <tr>
-   <td>70B
-   </td>
-   <td>140 GB
-   </td>
-   <td>70 GB
-   </td>
-   <td>35 GB
-   </td>
-  </tr>
-  <tr>
-   <td>405B
-   </td>
-   <td>810 GB
-   </td>
-   <td>405 GB
-   </td>
-   <td>203 GB
-   </td>
-  </tr>
-</table>
-
-_Note: These are estimated values and may vary based on specific implementation details and optimizations._
-
-Here are some notebooks to help you started:
-
-* Run Llama 8B in free Google Colab in half precision
-* [Run Llama 8B in 8-bits with bitsandbytes](./8bit_bnb.ipynb)
-* [Run Llama 8B in 4-bits with bitsandbytes](./4bit_bnb.ipynb)
-* [Run Llama 8B with AWQ & fused ops](./awq.ipynb)
+Would you like to run inference of the Llama 3.1 models locally?
+So do we! The memory requirements depend on the model size and the
+precision of the weights. Here's a table showing the approximate
+memory needed for different configurations:
+
+### Llama 3.1
+
+| Model Size | FP16 | FP8 | INT4(AWQ/GPTQ/bnb) |
+| :--: | :--: | :--: | :--: |
+| 8B | 16 GB | 8GB | 4GB |
+| 70B | 140 GB | 70GB | 35GB |
+|405B | 810 GB | 405GB | 204GB |
+
+### Llama 3.2
+| Model Size | BF16/FP16 | FP8 | INT4 |
+| :--: | :--: | :--: | :--: |
+| 1B | 2.5 GB | 1.25GB | 0.75GB |
+| 3B | 6.5 GB | 3.2GB | 1.75GB |
+
+> [!NOTE]
+> These are estimated values and may vary based on specific
+implementation details and optimizations.
+
+Working with the capable Llama 3.1 8B models:
+
+* [Run Llama 3.1 8B in 4-bits with bitsandbytes](./4bit_bnb.ipynb)
+* [Run Llama 3.1 8B in 8-bits with bitsandbytes](./8bit_bnb.ipynb)
+* [Run Llama 3.1 8B with AWQ & fused ops](./awq.ipynb)
+
+Working on the 🐘 big Llama 3.1 405B model:
+
 * [Run Llama 3.1 405B FP8](./fp8-405B.ipynb)
 * [Run Llama 3.1 405B quantized to INT4 with AWQ](./awq_generation.py)
 * [Run Llama 3.1 405B quantized to INT4 with GPTQ](./gptq_generation.py)
-* [Run assisted decoding with Llama 405B and Llama 8B](./assisted_decoding.py)
+
+## Model Fine Tuning:
+
+It is often not enough to run inference on the model, you would also
+need to fine tune the model on some custom dataset. We have got you
+covered.
+
+Fine tune models on your custom dataset:
```

Comment:
There is also https://github.com/huggingface/huggingface-llama-recipes/blob/main/sft_vlm.py for vision